scripts/action-handler.js

```diff
@@ -1382,29 +1380,43 @@ Hooks.once('tokenActionHudCoreApiReady', async (coreModule) => {
         // Get consume target and type
             const consumeId = item?.system?.consume?.target
             const consumeType = item?.system?.consume?.type
+            const consumeAmount = item?.system?.consume?.amount
 
-            if (consumeId === item.id) return ''
+            if (!consumeId || consumeId === item.id) return {}
 
             // Return resources
             if (consumeType === 'attribute') {
-                if (!consumeId) return ''
                 const parentId = consumeId.substr(0, consumeId.lastIndexOf('.'))
-                const target = this.actor.system[parentId]
+                const target = foundry.utils.getProperty(this.actor.system, parentId)
 
-                return (target) ? `${target.value ?? '0'}${(target.max) ? `/${target.max}` : ''}` : ''
-            }
+                if (target) {
+                    const text = `${target.value ?? '0'}${target.max ? `/${target.max}` : ''}`
+                    return {
+                        text,
+                        title : `${text} ${target.label ?? ''}`
+                    }
+                }
 
-            const target = this.items.get(consumeId)
+            } else {
 
-            // Return charges
-            if (consumeType === 'charges') {
-                const uses = target?.system.uses
+                const target = this.items.get(consumeId)
+
+                // Return charges
+                if (consumeType === 'charges') {
+                    return this.#getUsesData(target, target.name, consumeAmount)
+                }
 
-                return (uses?.value) ? `${uses.value}${(uses.max) ? `/${uses.max}` : ''}` : ''
+                // Return quantity
+                if (target?.system?.quantity) {
+                    const text = `${consumeAmount > 1 ? `${consumeAmount} ${coreModule.api.Utils.i18n('DND5E.of')} ` : ''}${target.system.quantity}`
```

Comment:
`text` isn't being used.

Reply:
Good catch, `text` was supposed to be used instead of the `text` attribute as it is now. Will fix.